admin/listPurchases/listView.html

```diff
@@ -146,36 +146,24 @@
     nav.innerHTML = subpageNavigation.join("");
 
     if(extraNavNeeded){
-        document.querySelector(`.subpage-selector-backwards`).addEventListener("click", function (e){
-            e.preventDefault();
-            if(currentPageIndex <= 0) return;
-            currentPageIndex--;
-            loadSomeOrders();
-        });
-        document.querySelector(`.subpage-selector-forward`).addEventListener("click", function (e){
-            e.preventDefault();
-            if(currentPageIndex >= amountOfSubpages - 1) return;
-            currentPageIndex++;
-            loadSomeOrders();
-        });
-        document.querySelector(`.subpage-selector-to-start`).addEventListener("click", function (e){
-            e.preventDefault();
-            currentPageIndex = 0;
-            loadSomeOrders();
-        });
-        document.querySelector(`.subpage-selector-to-end`).addEventListener("click", function (e){
-            e.preventDefault();
-            currentPageIndex = amountOfSubpages - 1;
-            loadSomeOrders();
-        });
+        document.querySelector(`.subpage-selector-backwards`).addEventListener("click", (e) =>
+            onClickSubpageSelectorGeneric(e, currentPageIndex > 0, () => currentPageIndex--)
+        );
+        document.querySelector(`.subpage-selector-forward`).addEventListener("click", (e) =>
+            onClickSubpageSelectorGeneric(e, currentPageIndex + 1 < amountOfSubpages, () => currentPageIndex++)
+        );
+        document.querySelector(`.subpage-selector-to-start`).addEventListener("click", (e) =>
+            onClickSubpageSelectorGeneric(e, () => currentPageIndex = 0)
+        );
+        document.querySelector(`.subpage-selector-to-end`).addEventListener("click", (e) =>
+            onClickSubpageSelectorGeneric(e, () => currentPageIndex = amountOfSubpages - 1)
+        );
```

Comment:
Like with the `createQuery...` methods I've commented on so far, your abstraction here doesn't actually *raise the abstraction level* which means it isn't very efficient in making the code better!

If you made a *more general* abstraction you could do that, and save on both complexity and code. Imagine this interface:

```js
modifyIndexOnClick(".subpage-selector-backwards", (currentIndex) => currentIndex > 0 ? currentIndex - 1 : currentIndex);
modifyIndexOnClick(".subpage-selector-forward", (currentIndex) => currentIndex + 1 < amountOfSubpages ? currentIndex + 1 : currentIndex);
modifyIndexOnClick(".subpage-selector-to-start", (currentIndex) => 0);
modifyIndexOnClick(".subpage-selector-to-end", (currentIndex) => amountOfSubpages - 1);
```

Inside `modifyIndexOnClick` you can check the value before and after modifying the page index, and only load orders if it has changed. And you can wrap all the boilerplate of both `document.querySelector(..).addEventListener` and `e.preventDefault`